sim/core/armor_test.go

```diff
@@ -0,0 +1,264 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/wowsims/wotlk/sim/core/stats"
+)
+
+func TestSunderArmorStacks(t *testing.T) {
+	sim := Simulation{}
+	baseArmor := 10643.0
+	target := Unit{
+		Type:         EnemyUnit,
+		Index:        0,
+		Level:        83,
+		auraTracker:  newAuraTracker(),
+		initialStats: stats.Stats{stats.Armor: baseArmor},
+		PseudoStats:  stats.NewPseudoStats(),
+		Metrics:      NewUnitMetrics(),
+	}
+	target.stats = target.initialStats
+	expectedArmor := baseArmor
+	if target.Armor() != expectedArmor {
+		t.Fatalf("Armor value for target should be %f but found %f", 10643.0, target.Armor())
+	}
+	stacks := int32(1)
+	sunderAura := SunderArmorAura(&target, stacks)
+	sunderAura.Activate(&sim)
+	tolerance := 0.001
+	for stacks <= 5 {
+		expectedArmor = baseArmor * (1.0 - float64(stacks)*0.04)
+		if target.Armor() < expectedArmor-tolerance || target.Armor() > expectedArmor+tolerance {
```

Comment:
This tolerance logic is used a lot. Could it be a new function in sim/core/utils.go, and maybe define tolerance just once as a const at the top of this file?

Reply:
Updated the PR to add this function and use it in the armor tests. I can refactor the other places in another PR.